tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,208 @@
+                        # during the interrupted sync.
+                        self.assertGreaterEqual(record_time, interrupted_stream_datetime)
+
+                        # Verify the interrupted sync replicates the expected record set all
+                        # interrupted records are in full records
+                        if primary_key:
+                            self.assertIn(record[primary_key], full_records_primary_keys,
+                                        msg="Incremental table record in interrupted sync not found in full sync")
+
+                    # Record count for all streams of interrupted sync match expectations
+                    records_after_interrupted_bookmark = 0
+                    for record in first_sync_stream_records:
+                        record_time = self.parse_date(record.get(replication_key))
+                        if record_time >= interrupted_stream_datetime:
+                            records_after_interrupted_bookmark += 1
+
+                    self.assertEqual(records_after_interrupted_bookmark, interrupted_record_count,
+                                    msg="Expected {} records in each sync".format(
+                                        records_after_interrupted_bookmark))
+
+                else:
+                    # Get the date to start 2nd sync for non-interrupted streams
+                    synced_stream_bookmark = interrupted_sync_state["bookmarks"].get(stream, dict()).get(replication_key, None)
+
+                    if synced_stream_bookmark:
+                        synced_stream_datetime = self.parse_date(synced_stream_bookmark)
+                    else:
+                        synced_stream_datetime = start_date_timestamp
+
+                    self.assertGreater(interrupted_record_count, 0,
+                                    msg="Un-interrupted streams must sync at least 1 record.")
+
+                    if replication_method == self.INCREMENTAL:
+
+                        for record in post_interrupted_sync_stream_records:
+                            record_time = self.parse_date(record.get(replication_key))
+
+                            # Verify resuming sync only replicates records with the replication key
+                            # values greater or equal to the state for streams that were replicated
+                            # during the interrupted sync.
+                            self.assertGreaterEqual(record_time, synced_stream_datetime)
+
+                            # Verify resuming sync replicates all records that were found in the full
+                            # sync (non-interrupted)
+                            self.assertIn(record, first_sync_stream_records,
+                                        msg="Unexpected record replicated in resuming sync.")
+                    else:
+                        # BUG: TDL-21675: interrupted sync does not sync already synced streams
```

Comment:
This may be expected behavior.  In this test set up bookmarks are advanced for all streams.  These streams may not have any data newer than the bookmark after the initial sync thus leaving the interrupted sync empty.  Based on the manipulated state and the data set it would be better to set expectations on which streams should replicate in each sync and how many if possible.
See for default case:
https://github.com/stitchdata/tap-tester/tree/master/tap_tester/base_suite_tests
See for more exhaustive coverage (5 interrupted sync tests):
https://github.com/singer-io/tap-google-ads/tree/main/tests